pkg/clients/tagging/v2/filters.go

```diff
@@ -0,0 +1,308 @@
+package v2
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/service/amp"
+	"github.com/aws/aws-sdk-go-v2/service/apigateway"
+	"github.com/aws/aws-sdk-go-v2/service/autoscaling"
+	"github.com/aws/aws-sdk-go-v2/service/databasemigrationservice"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/storagegateway"
+	"github.com/grafana/regexp"
+
+	"github.com/nerdswords/yet-another-cloudwatch-exporter/pkg/config"
+	"github.com/nerdswords/yet-another-cloudwatch-exporter/pkg/model"
+	"github.com/nerdswords/yet-another-cloudwatch-exporter/pkg/promutil"
+)
+
+type serviceFilter struct {
```

Comment:
Unfortunately I don't think it's possible 😢, the type definitions include the version specific client because we need access to the version specific AWS clients
https://github.com/kgeckhart/yet-another-cloudwatch-exporter/blob/18d830652adea4e3c4e75d6c3ce3887a6b438f07/pkg/clients/tagging/v2/filters.go#L32-L32

I struggled to try to find a sharable interface between the clients at this level since v1 and v2 have completely different paging implementations.